metadata/metadata.c

```diff
@@ -148,14 +151,30 @@ Allocator *create_allocator(VirtualDisk__ErasureCodeProfile ec_profile)
     return result;
 }
 
-Metadata *new_metadata()
+static void set_allocator(Allocator *allocator) {
+    metadata_disk_allocator = allocator;
+}
+
+Metadata *new_metadata(VirtualDisk__ErasureCodeProfile ec_profile)
 {
     Metadata *metadata = (Metadata *)malloc(sizeof(Metadata));
     metadata__init(metadata);
+    metadata->ec_profile = ec_profile;
```

Comment:
since we store the allocator in `metadata` we don't actually need it in the global static variable anymore

Reply:
sorry, we are not storing the allocator in metadata but only the profile with this PR. somehow we still need to ensure we are using the right allocator for the EC profile stored in metadata